### ISSUE TITLE
Move part of `catalog.newcontext.products` classes to QuerySet

### ISSUE THREAD

**Issue by duker33** [fidals/refarm-site#255]
This set of classes are not contexts. They don't have their own context data, but have only QuerySet logic.
Later we discussed this module using at site with the architect. And Artemiy said, that this classes are semantically separated from others in this module
https://github.com/fidals/shopelectro/pull/689#discussion_r248688141

If we'll move it's logic to QuerySets, we'll make this semantic more clear.

```
ActiveProducts
OrderedProducts
PaginatedProducts
ProductsByCategory
TaggedProducts
```

**Comment by duker33** [comment]
@artemiy312 , what do you think about this proposal?

**Comment by artemiy312** [comment]
Yes, we can do it. It looks like the django-way approach. 
Unfortunately it will lead to giant and imperative `QuerySet` classes, but it seems unavoidable when working with ORM and django  

Also we can remove some `tags` context classes in favor of `QuerySet` methods.
`PaginatedProducts` we should leave, because it isn't working only with `QuerySet`.

As the result only `PaginatedProducts, ProductImages, ProductBrands, GroupedTags` will remain.

**Comment by duker33** [comment]
@artemiy312 , we can organise this code in a right way. It'll take not more space, then it taken in contexts now.

PaginatedProducts does this two things:
-  just forwards `refarm_pagination.context.PaginationContext` class for context data.
- uses the class for queryset processing `refarm_pagination.context.PaginationContext`

Despite this two operations uses the same class, they are separated semantically. So, we can split them: the first peace we just throw away. The second one goes to queryset.

So, i'll propose some code

**Comment by duker33** [comment]
i'll adapt SE to the new refarm's code in the same task

**Comment by 0pdd** [comment]
@duker33 the puzzle [#266](https://github.com/fidals/refarm-site/issues/266) is still not solved.

**Comment by 0pdd** [comment]
@duker33 the puzzle [#308](https://github.com/fidals/refarm-site/issues/308) is still not solved; solved: [#266](https://github.com/fidals/refarm-site/issues/266).
